src/high_level.rs

```diff
@@ -616,24 +617,43 @@ pub fn look_for_dep_with_data_hash(data_hash: &[u8]) -> Result<usize, SysError>
     look_for_dep_with_hash2(data_hash, ScriptHashType::Data)
 }
 
+pub fn encode_hex(data: &[u8]) -> CString {
+    let mut s = String::with_capacity(data.len() * 2);
+    for &b in data {
+        write!(&mut s, "{:02x}", b).unwrap();
+    }
+    CString::new(s).unwrap()
+}
+
+pub fn decode_hex(data: &CStr) -> Vec<u8> {
+    let data = data.to_str().unwrap();
+    (0..data.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&data[i..i + 2], 16).unwrap())
+        .collect()
+}
```

Comment:
should check string length and if contains hex char only.

Reply:
If `data` is not a legal hex string, this function will panic.